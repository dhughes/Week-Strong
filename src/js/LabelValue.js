import React from 'react';
import { Hbox } from './Box';
import styled from 'styled-components';

const LeftDiv = styled.div`
  text-align: left;
`;

const RightDiv = styled.div`
  text-align: right;
`;

const LabelValue = props => {
  return (
    <Hbox>
      <LeftDiv>{props.label}:</LeftDiv>
      <RightDiv>{props.value}</RightDiv>
    </Hbox>
  );
};

export default LabelValue;
