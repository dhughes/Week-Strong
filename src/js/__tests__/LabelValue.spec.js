import React from 'react';
import { shallow, mount } from 'enzyme';
import LabelValue from '../LabelValue';

xtest('LabelValue renders', () => {
  const component = shallow(<LabelValue />);
  expect(component).toMatchSnapshot();
});

test('LabelValue shows Test label', () => {
  const component = mount(<LabelValue label="Test" />);
  expect(component.text().indexOf('Test')).toBeGreaterThan(-1);
});

test('LabelValue shows Streak label', () => {
  const component = mount(<LabelValue label="Streak" />);
  expect(component.text().indexOf('Streak')).toBeGreaterThan(-1);
});

test('LabelValue shows 100 value', () => {
  const component = mount(<LabelValue label="Test" value={100} />);
  expect(component.text().indexOf('100')).toBeGreaterThan(-1);
});

test('LabelValue shows 200 value', () => {
  const component = mount(<LabelValue label="Test" value={200} />);
  expect(component.text().indexOf('200')).toBeGreaterThan(-1);
});

test('LabelValue shows non-numeric values', () => {
  const component = mount(<LabelValue label="Streak" value="8 days" />);
  expect(component.text().indexOf('8 days')).toBeGreaterThan(-1);
});
