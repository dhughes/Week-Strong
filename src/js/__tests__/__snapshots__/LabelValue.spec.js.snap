// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelValue renders 1`] = `
<div>
  Test
</div>
`;
